src/app/pages/tasks/component.ts

```diff
@@ -58,9 +71,30 @@ export class TasksComponent implements OnInit {
     }
     `;
 
-    this.api
-      .query(query)
-      .subscribe((response) => this.setTaskData(response.data));
+    this.api.query(query).subscribe((response) => {
+      this.setTaskData(response.data);
+
+      this.taskScores = Object.keys(this.taskTypeFilters).map(
+        (function (obj: any) {
+          function getTask(taskType: string) {
+            const filtered = obj.filterByTaskType(taskType);
+            return {
+              descriptionShort: taskType,
+              earnedPoints: filtered.reduce(
+                (acc, x) => x.earnedPoints + acc,
+                0
+              ),
+              value: filtered.reduce((acc, x) => x.value + acc, 0),
+            };
+          }
+          return getTask;
+        })(this)
+      );
+    });
+  }
+
+  private filterByTaskType(taskType: string) {
```

Comment:
```suggestion
  private filterTasks(taskType: string) {
```

"ByTaskType" is implied by the argument